cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -150,3 +206,56 @@ func (crs *containerRegistryService) createRegistriesClient(
 
 	return client, nil
 }
+
+func (crs *containerRegistryService) getAcrToken(
+	ctx context.Context,
+	subscriptionId string,
+	loginServer string,
+) (*acrToken, error) {
+	creds, err := crs.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("getting credentials for subscription '%s': %w", subscriptionId, err)
+	}
+
+	token, err := creds.GetToken(ctx, policy.TokenRequestOptions{Scopes: auth.LoginScopes})
+	if err != nil {
+		return nil, fmt.Errorf("getting token for subscription '%s': %w", subscriptionId, err)
+	}
+
+	formData := url.Values{}
+	formData.Set("grant_type", "access_token")
+	formData.Set("service", loginServer)
+	formData.Set("access_token", token.Token)
+
+	tokenUrl := fmt.Sprintf("https://%s/oauth2/exchange", loginServer)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, tokenUrl, strings.NewReader(formData.Encode()))
+	if err != nil {
+		return nil, fmt.Errorf("creating request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := crs.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("getting token: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading response body: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
If we do use `azruntime.Pipeline` here, the default `err` has more details around the HTTP failure that will be useful for diagnosing, and we don't have to create a custom err message